src/main/java/bio/terra/service/snapshot/flight/create/SnapshotCreateFlight.java

```diff
@@ -314,14 +313,13 @@ public SnapshotCreateFlight(FlightMap inputParameters, Object applicationContext
     addStep(new SnapshotAuthzIamStep(iamClient, snapshotService, snapshotReq, userReq, snapshotId));
 
     // Now that the snapshot exists in Sam, we can add data access control groups to the snapshot
-    List<String> dataAccessControlGroups = snapshotReq.getDataAccessControlGroups();
-    if (Objects.nonNull(dataAccessControlGroups) && !dataAccessControlGroups.isEmpty()) {
-      addStep(
-          new CreateSnapshotGroupConstraintPolicyStep(
-              policyService, snapshotId, dataAccessControlGroups));
-      addStep(
-          new AddSnapshotAuthDomainStep(iamService, userReq, snapshotId, dataAccessControlGroups));
-    }
+    addStep(new CreateSnapshotSetDataAccessGroupsStep(snapshotReq.getDataAccessControlGroups()));
+    addStep(
+        new IfDataAccessControlGroupStep(
```

Comment:
why change the if to a step? Is this the direction we want to move in/ the way to follow flight guidelines?